src/main/java/bio/terra/service/snapshotbuilder/utils/HierarchyQueryBuilder.java

```diff
@@ -38,22 +38,31 @@ public Query generateQuery(SnapshotBuilderDomainOption domainOption, int concept
     var conceptRelationship =
         TableVariable.forPrimary(TablePointer.fromTableName(ConceptRelationship.TABLE_NAME));
     var relationshipId = conceptRelationship.makeFieldVariable(ConceptRelationship.RELATIONSHIP_ID);
-    var conceptId1 = conceptRelationship.makeFieldVariable(ConceptRelationship.CONCEPT_ID_1);
-    var conceptId2 = conceptRelationship.makeFieldVariable(ConceptRelationship.CONCEPT_ID_2);
+    var parentId = conceptRelationship.makeFieldVariable(ConceptRelationship.CONCEPT_ID_1);
+    var childId = conceptRelationship.makeFieldVariable(ConceptRelationship.CONCEPT_ID_2);
     var child =
         TableVariable.forJoined(
-            TablePointer.fromTableName(Concept.TABLE_NAME), Concept.CONCEPT_ID, conceptId2);
+            TablePointer.fromTableName(Concept.TABLE_NAME), Concept.CONCEPT_ID, childId);
     var parent =
         TableVariable.forJoined(
-            TablePointer.fromTableName(Concept.TABLE_NAME), Concept.CONCEPT_ID, conceptId1);
+            TablePointer.fromTableName(Concept.TABLE_NAME), Concept.CONCEPT_ID, parentId);
     FieldVariable conceptName = child.makeFieldVariable(Concept.CONCEPT_NAME);
     FieldVariable conceptCode = child.makeFieldVariable(Concept.CONCEPT_CODE);
 
+    // To get the total occurrence count for a child concept, we need to join the child through the
+    // ancestor table to find all of its children. We don't need to use a left join here
+    // because every concept has itself as an ancestor, so there will be at least one match.
+    var conceptAncestor =
+        TableVariable.forJoined(
+            TablePointer.fromTableName(ConceptAncestor.TABLE_NAME),
+            ConceptAncestor.ANCESTOR_CONCEPT_ID,
+            childId);
+
     TableVariable domainOccurrence =
```

Comment:
This is done in `SearchConceptsQueryBuilder` [here](https://github.com/DataBiosphere/jade-data-repo/blob/8d5cadf6492aafe3774ebbf9be58ce0d6c722fe0/src/main/java/bio/terra/service/snapshotbuilder/utils/SearchConceptsQueryBuilder.java#L57). The only difference is 

'conceptAncestor.makeFieldVariable(ConceptAncestor.DESCENDANT_CONCEPT_ID' is inlined. 

Thoughts on making a helper function?

Reply:
Let me take a look. I was thinking it could wait for a future PR when we do a larger refactor on this code, but maybe it's small enough to do this part now.

Reply:
I think this code is duplicated in three places, and I think the best fix would be to refactor into a class that encapsulates a concept occurrence table. That's enough work for a separate ticket so I created https://broadworkbench.atlassian.net/browse/DC-1031 for it

Reply:
Thank you!